### PR TITLE
Code refactoring

### LINKED ISSUE #55: Refactor the code
Estimated time: 3h
Real time: 2h45

### PR DESCRIPTION
**Pull request by clemdiep-heig**
close #55

### CODE REVIEW

File: src/main/java/ch/heigvd/statique/convertors/Builder.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  /**
+   * Explores a directory and creates a build directory When the explorer sees a
+   * directory, it recursively explores the directory and copies it When the
+   * explorer sees a file, it converts the file or copy it in the build
+   * directory.
+   *
+   * @param rootDirectory  root directory
+   * @param buildDirectory build directory
```
Comment (dylanrodr-heig) [inline src/main/java/ch/heigvd/statique/convertors/Builder.java:79]:
```suggestion
   * @param buildDestination build directory
```

File: src/main/java/ch/heigvd/statique/utils/Config.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  public Map<String, Object> toRender() {
+    return config;
+  }
+
+  /**
+   * Merge this config with another one. This will not affect the original
+   * config. If a key is present in both configs, the value of the other config
+   * will be used.
+   *
+   * @param config The config to merge with.
```
Comment (dylanrodr-heig) [inline src/main/java/ch/heigvd/statique/utils/Config.java:40]:
```suggestion
   * @param other The config to merge with.
```

File: src/main/java/ch/heigvd/statique/utils/Page.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   * @return The file content.
+   */
+  private String readFile() throws IOException {
+    return FileUtils.readFileToString(fromPath.toFile(), StandardCharsets.UTF_8);
+  }
+
+  /**
+   * Write the HTML to the destination.
+   *
+   * @param html The HTML to write.
```
Comment (dylanrodr-heig) [inline src/main/java/ch/heigvd/statique/utils/Page.java:48]:
```suggestion
   * @param content The HTML to write.
```

File: src/main/java/ch/heigvd/statique/utils/Page.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   *         is the markdown.
+   */
+  private String[] separateYamlMd(String fileContent) {
+    return new String[] {"", fileContent};
+  }
+
+  /**
+   * Convert the markdown to HTML and apply the template.
+   *
+   * @param markdown The markdown to convert.
```
Comment (dylanrodr-heig) [inline src/main/java/ch/heigvd/statique/utils/Page.java:68]:
```suggestion
   * @param md The markdown to convert.
```

**Review by dylanrodr-heig — CHANGES_REQUESTED** [review]
It's quite perfect! Just need some changes.
You can also update the package `org.commonmark` at line 70 in the file `pom.xml` to `0.18.2`.

**Review by dylanrodr-heig — APPROVED** [review]

**Review by Isotech42 — APPROVED** [review]
Wow, quite a cleanup in there. It really needed to be done at some point. Thanks for the work!

**Pull request merged**